### DISCUSSION TITLE
How to pass data from the context of an activity to a child's workflow

### DISCUSSION THREAD

**Discussion by zhangtingwz** [elsa-workflows/elsa-core#3351] [Q&A]
Hello, it's a pleasure to meet you, as well as using elsa workflow, in the process of using it, I encountered a problem：
![image](https://user-images.githubusercontent.com/26179617/194691202-0f0dac9a-e5e1-4c35-9045-4a6b053ceff4.png)
but 
![image](https://user-images.githubusercontent.com/26179617/194691246-c895c28d-e362-4c26-a24e-d92709cefe54.png)
please help me ，thank you very much

**Comment by zhangtingwz** [comment]
This is just an example, mainly to represent the values I want from the activity
![image](https://user-images.githubusercontent.com/26179617/194691628-32a79b8e-eb69-41f3-a25f-7bf4d23adc81.png)

**Comment by IbrahimMNada** [comment]
this is not an issue this is more like a question.. 

when you specify a `input `into the `RunWorkflow `method.  thats an input for the new workflow instance wide. 
so when you set a vibrable on the next child workflow activity. notice the `context`  is of type `ActivityExecutionContext`,
so its not workflow wide it's scoped within the activity you are on. so, if you want to get a variable that is  workflow wide use this `context.WorkflowInstance.Variables` i think you will fide you dto there as json. 

and if u have more question here is the discord link for that 
https://discord.gg/F5tCv2bs

cheers

**Comment by zhangtingwz** [comment]
Yes, my friend, this is really a problem. Because of the wall, I can't go to Stack Overflow. Then elsa is a new workflow, and there are fewer people using it. So I can only go to github to find the answer. I'm sorry for causing you trouble, and thank you for your help

**Reply by IbrahimMNada** [reply]
no problem man, 
you can ask at anytime. 
I suggest asking on discord is the best/fastest way 
https://discord.gg/F5tCv2bs

**Comment by sfmskywalker** [comment]
@zhangtingwz the reason the `fillingEntityDto` value is null in the child workflow is because at the time your workflow setup executed, it took the current value of that field which is `null` at that time (since your workflow has yet to execute).
Another issue is that the field isn't stored persistently as part of the workflow instance.

The first thing to do is convert `fillingEntityDto` to be a property that uses `GetState` and `SetState`, like this:

```csharp
private FillingEntityDto FillingEntityDto
{
   get => GetState();
   set => SetState(value);
}
```

Then you need to provide the input "lazily" using a lambda, like this:

```csharp
.RunWorkflow<RegistCenterVerifyChildWorkflow>(RunWorkflowMode.Blocking, context => FillingEntityDto);
```

Then it should work.
